pandas/core/arrays/datetimelike.py

```diff
@@ -526,6 +528,13 @@ def __getitem__(self, key):
         only handle list-likes, slices, and integer scalars
         """
 
+        if lib.is_integer(key):
+            # fast-path
+            result = self._data[key]
+            if self.ndim == 1:
+                return self._box_func(result)
+            return self._simple_new(result, dtype=self.dtype)
+
```

Comment:
elif

Reply:
i think i prefer it this way.  the if/elif/else cases that follow this fit together nicely, dont return-early